inception-project/inception
# #3969 - Review and improve visibility calculation code

**What's in the PR**
- Switch various variables to `var`
- Move code hiding a suggestion based on a learning record status to a central place
- Removed some redundant parameters from methods
- Merge LearningRecordService into the RecommendationService
- Add a write-through cache for the learning records to the recommender state

**How to test manually**
* Check if recommenders work properly
* Check if active learning works properly
* Check both of the above as a normal annotator and also while using sidebar curation mode

**Automatic testing**

**Documentation**
